### PR TITLE
Fixes Reagent Dispensers Not Removing Reagents

### LINKED ISSUE #67329: dragging water/fuel tanks doesn't decrease reagent count
## Testmerges:

irrelevant

## Reproduction:

> drag around tank of fuel
wrench it
now it is leaving piles of welder fuel behind it
examine fuel tank
still full of welding fuel

seems like an oversight

### PR DESCRIPTION
**Pull request by 13spacemen**
Fixes https://github.com/tgstation/tgstation/issues/67329
tested and works with fuel, large fuel, foam, water, large water, etc.
stops leaking reagents at  <amount_to_leak

:cl:
fix: Reagent dispensers will actually remove reagents upon leaking
/:cl:

### CODE REVIEW

File: code/modules/reagents/reagent_dispenser.dm [hunk 1]
```diff
@@ -19,6 +19,8 @@
 	var/openable = FALSE
 	///Is this dispenser slowly leaking its reagent?
 	var/leaking = FALSE
+	//volume modifier for reagent procs
```
Comment (tf-4) [inline code/modules/reagents/reagent_dispenser.dm:22]: Autodoc

File: code/modules/reagents/reagent_dispenser.dm [hunk 2]
```diff
@@ -19,6 +19,8 @@
...
+	var/volumemod
```
Comment (tf-4) [inline code/modules/reagents/reagent_dispenser.dm:23]:
```suggestion
	var/volume_mod
```

File: code/modules/reagents/reagent_dispenser.dm [hunk 3]
```diff
@@ -75,22 +77,25 @@
 	else
 		qdel(src)
 
+/obj/structure/reagent_dispensers/proc/tank_leak()
+	if(leaking && reagents && reagents.total_volume >= amount_to_leak)
+		reagents.expose(get_turf(src), TOUCH, amount_to_leak / max(amount_to_leak, reagents.total_volume))
+		reagents.remove_reagent(reagent_id, amount_to_leak)
+		return TRUE
```
Comment (Mothblocks) [inline code/modules/reagents/reagent_dispenser.dm:84]:
```suggestion
		return TRUE
	return FALSE
```

```suggestion
```

Generally, everything should return a value or shouldn't return a value. Pick one.

File: code/modules/reagents/reagent_dispenser.dm [hunk 4]
```diff
@@ -75,22 +77,25 @@
...
-		reagents.expose(get_turf(src), TOUCH, 10 / max(10, reagents.total_volume))
+	tank_leak()
 	return TOOL_ACT_TOOLTYPE_SUCCESS
 
 /obj/structure/reagent_dispensers/Moved(atom/OldLoc, Dir)
 	. = ..()
-	if(leaking && reagents)
-		reagents.expose(get_turf(src), TOUCH, 10 / max(10, reagents.total_volume))
-
+	tank_leak()
```
Comment (Mothblocks) [inline code/modules/reagents/reagent_dispenser.dm:98]:
```suggestion
	tank_leak()

```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Pull request merged**